internal/cmd/schema-registry/schema_utilities.go

```diff
@@ -73,6 +86,42 @@ func ReadSchemaRefs(cmd *cobra.Command) ([]srsdk.SchemaReference, error) {
 	return refs, nil
 }
 
+func readMetadata(flagName string, cmd *cobra.Command) (*srsdk.Metadata, error) {
+	var metadata srsdk.Metadata
+	metadataPath, err := cmd.Flags().GetString(flagName)
+	if err != nil {
+		return nil, err
+	}
+	if metadataPath != "" {
+		metadataBlob, err := os.ReadFile(metadataPath)
+		if err != nil {
+			return nil, err
+		}
+		if err = json.Unmarshal(metadataBlob, &metadata); err != nil {
+			return nil, err
+		}
+	}
+	return &metadata, nil
+}
+
+func readRuleset(flagName string, cmd *cobra.Command) (*srsdk.RuleSet, error) {
```

Comment:
Can we read the flag outside of this function, so we don't need to pass `cmd`? It'll be easier to unit test that way.